### PR TITLE
implement PrepareDistortedPix as part of DegradeImage

### PR DESCRIPTION
**Pull request by Shreeshrii**
Fixes https://github.com/tesseract-ocr/tesseract/issues/1052

### CODE REVIEW

**Comment by amitdo** [comment]
Shree, I don't think mixing the distort & degrade methods is right.
 
Basically, the distort method should be directly called from text2image.cpp.

File: src/training/degradeimage.cpp [hunk 1]
```diff
@@ -210,6 +215,8 @@ Pix* PrepareDistortedPix(const Pix* pix, bool perspective, bool invert,
         (*boxes)[b].set_right((*boxes)[b].left() + 1);
     }
   }
+  if (invert && randomizer->SignedRand(1.0) < -0.9)
```
[the remark this answers is not in the document]
Comment (amitdo) [reply src/training/degradeimage.cpp]: Also,
* Todo -> TODO (like in other places in the codebase).
* Why are you keeping the commented code in DegradeImage()?

[the remark this answers is not in the document]
Comment (Shreeshrii) [reply src/training/degradeimage.cpp]: >0 (50% percents of images inverted).

Is that a realistic expectation of the types of images which will be OCRed?

[the remark this answers is not in the document]
Comment (Shreeshrii) [reply src/training/degradeimage.cpp]: * Todo -> TODO (like in other places in the codebase).

OK.

* Why are you keeping the commented code in DegradeImage()?

I meant to delete it after the changes to text2image worked fine. I will fix that.

**Comment by amitdo** [comment]
> 0 (50% percents of images inverted).
>
>Is that a realistic expectation of the types of images which will be OCRed?

Generally no, but...
If you want good accuracy for inverted images, you should train with enough samples.

**Comment by Shreeshrii** [comment]
>If you want good accuracy for inverted images, you should train with enough samples.

OK, that makes sense. I will change it to 0.

Thanks!

**Comment by amitdo** [comment]
>Thanks!

Thank you too :-)

**Pull request merged**